AdyenNetworking/Helpers/Logging/DebugLogger.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  DebugLogger.swift
+//  AdyenNetworking
+//
+//  Created by Alexander Guretzki on 26/02/2025.
+//
+
+import Foundation
+
+internal struct DebugLogger: DebugLogging {
+    
+    func print(_ items: Any..., separator: String, terminator: String, fileId: String) {
+        guard Logging.isEnabled else { return }
```

Comment:
Did you consider passing something like `DummyLogger` and moving this check to the assembly layer?

Reply:
Due to the current implementation of `Logging` acting basically as a singleton this might not reproduce the current behavior where you can toggle logging any time during the lifecycle of the app.
I believe we don't use it like that but it's breaking behavior.

I'd instead like to get rid of the `Logging` enum in a future major release, what do you think?

Reply:
I agree. Ideally, as a developer, I'd like to have an option to provide my own logger to a networking framework upon initialization.